itou/users/models.py

```diff
@@ -321,12 +319,19 @@ class Meta(AbstractUser.Meta):
             # Make sure that if you have a lack_of_nir_reason value, you cannot have a nir value
             # (but we'll have a lot of users lacking both nir & lack_of_nir_reason values)
             models.CheckConstraint(
-                check=Q(lack_of_nir_reason="") | Q(nir__isnull=True),
+                check=Q(lack_of_nir_reason="") | Q(nir=""),
                 name="user_lack_of_nir_reason_or_nir",
                 violation_error_message=(
                     "Un utilisateur ayant un NIR ne peut avoir un motif justifiant l'absence de son NIR."
                 ),
             ),
+            UniqueConstraintWithErrorCode(
+                "nir",
+                name="unique_nir_if_not_empty",
+                validation_error_code="unique_nir_if_not_empty",
+                condition=~Q(nir=""),  # XXX: Should we only apply this constraint on job seekers ?
```

Comment:
nit : on peut grouper les `validation_error_*`.
```suggestion
                condition=~Q(nir=""),
                validation_error_code="unique_nir_if_not_empty",
                  # XXX: Should we only apply this constraint on job seekers ?
```